osquery/events/darwin/endpointsecurity_fim.cpp

```diff
@@ -0,0 +1,212 @@
+    }
+
+    auto es_sub = es_subscribe(es_file_client_, &events[0], events.size());
+    if (es_sub != ES_RETURN_SUCCESS) {
+      VLOG(1) << "Couldn't subscribe to EndpointSecurity subsystem";
+    }
+  }
+}
+
+bool EndpointSecurityFileEventPublisher::shouldFire(
+    const EndpointSecurityFileSubscriptionContextRef& sc,
+    const EndpointSecurityFileEventContextRef& ec) const {
+  return true;
+}
+
+void EndpointSecurityFileEventPublisher::tearDown() {
+  if (es_file_client_ == nullptr) {
+    return;
+  }
+  es_unsubscribe_all(es_file_client_);
+  if (es_file_client_success_) {
+    auto result = es_delete_client(es_file_client_);
+    if (result != ES_RETURN_SUCCESS) {
+      VLOG(1) << "endpointsecurity_fim: error tearing down es_client";
+    }
+    es_file_client_ = nullptr;
+  }
+}
+
+void EndpointSecurityFileEventPublisher::handleMessage(
+    const es_message_t* message) {
+  if (message == nullptr) {
+    return;
+  }
+
+  if (message->action_type == ES_ACTION_TYPE_AUTH) {
+    return;
+  }
+  auto ec = createEventContext();
+  ec->version = message->version;
+  if (ec->version >= 2) {
+    ec->seq_num = message->seq_num;
+  }
+
+  if (ec->version >= 4) {
+    ec->global_seq_num = message->global_seq_num;
+  }
+
+  getProcessProperties(message->process, ec);
+
+  switch (message->event_type) {
+  case ES_EVENT_TYPE_NOTIFY_CREATE: {
+    ec->event_type = "create";
+    if (message->event.create.destination_type ==
+        ES_DESTINATION_TYPE_EXISTING_FILE) {
+      ec->filename = getStringFromToken(
+          &message->event.create.destination.existing_file->path);
+    } else {
+      std::string filename = getStringFromToken(
+          &message->event.create.destination.new_path.dir->path);
+      filename += '/';
+      filename += getStringFromToken(
+          &message->event.create.destination.new_path.filename);
+      ec->filename = filename;
+    }
+  } break;
+  case ES_EVENT_TYPE_NOTIFY_WRITE: {
+    ec->event_type = "write";
+    ec->filename = getStringFromToken(&message->event.write.target->path);
+  } break;
+  case ES_EVENT_TYPE_NOTIFY_RENAME: {
+    ec->event_type = "rename";
+    ec->filename = getStringFromToken(&message->event.rename.source->path);
+    if (message->event.rename.destination_type ==
+        ES_DESTINATION_TYPE_EXISTING_FILE) {
+      ec->dest_filename = getStringFromToken(
+          &message->event.rename.destination.existing_file->path);
+    } else {
+      std::string filename = getStringFromToken(
+          &message->event.rename.destination.new_path.dir->path);
+      filename += '/';
+      filename += getStringFromToken(
+          &message->event.rename.destination.new_path.filename);
+      ec->dest_filename = filename;
+    }
+  } break;
+  case ES_EVENT_TYPE_NOTIFY_TRUNCATE: {
+    ec->event_type = "truncate";
+    ec->filename = getStringFromToken(&message->event.truncate.target->path);
+  } break;
+  default:
+    VLOG(1) << "endpointsecurity_fim: unexpected event";
```

Comment:
Can we also log the event_type?

Reply:
I added the `event_type` here in the log, but it's also an enum, so we may not get the name of it. But still useful I guess